apps/trading/components/market-banner/market-update-state-banner.tsx

```diff
@@ -24,6 +24,16 @@ export const MarketUpdateStateBanner = ({
   const t = useT();
   const governanceLink = useLinks(DApp.Governance);
 
+  const openTradingProposals = sortBy(
+    proposals.filter(
+      (p) =>
+        p.state === ProposalState.STATE_OPEN &&
+        p.terms.change.__typename === 'UpdateMarketState' &&
+        market.state === MarketState.STATE_SUSPENDED_VIA_GOVERNANCE
```

Comment:
In order to get the actual resuming proposals it'd need to have additional condition:
```suggestion
        market.state === MarketState.STATE_SUSPENDED_VIA_GOVERNANCE &&
        p.terms.change.updateType === MarketUpdateType.MARKET_STATE_UPDATE_TYPE_RESUME
```

Reply:
Additionally I'd recheck if `useUpdateMarketStateProposals` which is used one level up from here is returning the actual proposal data or the whole **batch** proposal (if the update market state is inside a batch).

If it is batch then we'd need to change the hook to handle proposal picking from `proposal.proposals` which is an array of proposals (they have its own id and `batchId` which should be used for the link in this case)

Reply:
from what I can see `useUpdateMarketStateProposals` returns `MarketViewProposalFieldsFragment[]`  which is the actual proposal data and there's no `batchId`

Reply:
I have updated the queries to flatten the proposals